### PR TITLE
Mysadnaangular11 - Transfer to `startup-os` Firebase DB

### PR DESCRIPTION
**Pull request by a3l5o7n9**
The app now gets data from the Firebase DB `startup-os`, and no longer uses mock data.

### CODE REVIEW

File: projects/mysadna/src/app/components/project-details/project-details.component.ts [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { zip } from 'rxjs';
```
Comment (webdevelopland) [inline projects/mysadna/src/app/components/project-details/project-details.component.ts]: :+1:

File: projects/mysadna/src/app/services/firebase.service.ts [hunk 2]
```diff
@@ -3,7 +3,8 @@ import { AngularFirestore, AngularFirestoreCollection } from 'angularfire2/fires
 import { Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
 
-import { Project, User } from '@/proto';
+// import { Project, User } from '@/proto';
```
Comment (webdevelopland) [inline projects/mysadna/src/app/services/firebase.service.ts]: commented code :scream:

File: projects/mysadna/src/app/components/user-details/user-details.component.html [hunk 3]
```diff
@@ -4,16 +4,18 @@
...
-              <img src="./assets/images/{{socialNetwork}}.svg">
-            </a>
+            <div class="social-network" *ngFor="let socialNetwork of user.getSocialNetworkList()">
+              <a [href]="socialNetwork.getUrl()">
+                <img [src]="'./assets/images/' + types[socialNetwork.getType()] + '.svg'">
+              </a>
+            </div>
+          </div>
+          <div class="cv-url">
+            <a href="https://my-first-firebase-projec-be6b0.firebaseapp.com/cv-template.html/">CV File</a>
```
Comment (webdevelopland) [inline projects/mysadna/src/app/components/user-details/user-details.component.html]: It's not a mistake, but why absolute link?
Why not:
```html
<a href="cv-template.html/">CV File</a>
```
Thing is, if we'll decide to deploy app on different domain (e.g. [https://mysadna-profile-pages.firebaseapp.com](https://mysadna-profile-pages.firebaseapp.com))
We don't need to change the link.

Comment (webdevelopland) [reply projects/mysadna/src/app/components/user-details/user-details.component.html]: Also, I think the link is incorrect. It should be
```
assets/cv-pages/cv-template.html
```
isn't it?

File: projects/mysadna/proto/data.proto [hunk 4]
```diff
@@ -4,37 +4,100 @@ package com.hasadna.mysadna;
...
+  string id = 1;
+  string display_name = 2;
+  string project_url = 3;
+  // Short description
+  string description = 4;
+  // This is redundant since User lists projects, but we don't want to do the
+  // calculation in the front-end. We might add a test to enforce this, or push
+  // a better data structure to Firebase.
+  repeated string user_id = 5;
+}
```
Comment (webdevelopland) [inline projects/mysadna/proto/data.proto]: No end with new line, but it's ok. We're not going to use the file.

File: projects/mysadna/src/app/components/user-details/user-details.component.ts [hunk 5]
```diff
@@ -13,30 +12,29 @@ import { FirebaseService } from '@/services';
 export class UserDetailsComponent implements OnInit {
   user: User;
   userProjects: Project[];
+  types: string[] = Object.keys(SocialNetwork.Type);
```
Comment (webdevelopland) [inline projects/mysadna/src/app/components/user-details/user-details.component.ts]: :+1:

File: projects/mysadna/src/app/components/user-details/user-details.component.html (hunk 3 above)
Comment (a3l5o7n9) [reply projects/mysadna/src/app/components/user-details/user-details.component.html]: I'll try that.

File: projects/mysadna/src/app/services/firebase.service.ts (hunk 2 above)
Comment (a3l5o7n9) [reply projects/mysadna/src/app/services/firebase.service.ts]: Removed.

File: projects/mysadna/proto/data.proto (hunk 4 above)
Comment (a3l5o7n9) [reply projects/mysadna/proto/data.proto]: Should I remove the file and its folder, then?

Comment (webdevelopland) [reply projects/mysadna/proto/data.proto]: Not yet, we need to start using new protoc first.

**Pull request merged**